core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -224,19 +225,15 @@ object EntityDecoder {
     text.map(_.toArray)
 
   // File operations
-  def binFile[F[_]](file: File, blocker: Blocker)(implicit
-      F: Sync[F],
-      cs: ContextShift[F]): EntityDecoder[F, File] =
+  def binFile[F[_]: Files: Concurrent](file: File): EntityDecoder[F, File] =
     EntityDecoder.decodeBy(MediaRange.`*/*`) { msg =>
-      val pipe = writeAll[F](file.toPath, blocker)
+      val pipe = Files[F].writeAll(file.toPath)
       DecodeResult.success(msg.body.through(pipe).compile.drain).map(_ => file)
     }
 
-  def textFile[F[_]](file: File, blocker: Blocker)(implicit
-      F: Sync[F],
-      cs: ContextShift[F]): EntityDecoder[F, File] =
+  def textFile[F[_]: Files](file: File)(implicit F: Sync[F]): EntityDecoder[F, File] =
```

Comment:
why `Sync` here? Also if `Sync` is unavoidable, you can remove `Files` from there

Reply:
There's a single call that needs to be suspended downstream. I'll try to see if I can eliminate that.